File: plugin/src/main/kotlin/io/nimbly/tzatziki/util/JavaUtil.java
package io.nimbly.tzatziki.util;

import java.lang.reflect.Field;

public class JavaUtil {

    public static void updateField(Object object, String field, boolean bool) {
        try {
            Field f1 = object.getClass().getDeclaredField(field);
            f1.setAccessible(true);
            f1.setBoolean(object, bool);
        } catch (Exception ignored) {
        }
    }

    public static void updateField(Object object, String field, int inte) {
        try {
            Field f1 = object.getClass().getDeclaredField(field);
            f1.setAccessible(true);
            f1.setInt(object, inte);
        } catch (Exception ignored) {
        }
    }
}

File: i18n/src/main/kotlin/io/nimbly/i18n/util/JavaUtil.kt
package io.nimbly.i18n.util

object JavaUtil {

    fun updateField(`object`: Any, field: String?, str: String) {
        try {
            val f1 = `object`.javaClass.getDeclaredField(field)
            f1.isAccessible = true
            f1.set(`object`, str)
        } catch (ignored: Exception) {
        }
    }

    fun updateField(`object`: Any, field: String?, bool: Boolean) {
        try {
            val f1 = `object`.javaClass.getDeclaredField(field)
            f1.isAccessible = true
            f1.setBoolean(`object`, bool)
        } catch (ignored: Exception) {
        }
    }

    fun updateField(`object`: Any, field: String?, inte: Int) {
        try {
            val f1 = `object`.javaClass.getDeclaredField(field)
            f1.isAccessible = true
            f1.setInt(`object`, inte)
        } catch (ignored: Exception) {
        }
    }
}
